src/lib.rs

```diff
@@ -628,6 +628,16 @@ impl Config {
                 cmd.arg("-DCMAKE_OSX_SYSROOT=/");
                 cmd.arg("-DCMAKE_OSX_DEPLOYMENT_TARGET=");
             }
+        } else if target.contains("darwin") {
+            if !self.defined("CMAKE_OSX_ARCHITECTURES") {
+                if target.contains("x86_64") {
```

Comment:
This will be broken for `x86_64h-apple-darwin` if Rust ever adds it in the future (clang has this target, for example). This is a note to self as much as anything, and there's no need to handle hypothetical targets.